network/p2p/gossip/mempool.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+var ErrDuplicateTx = errors.New("duplicate tx")
+
+type Mempool[T Gossipable] struct {
```

Comment:
Since we're exposing a new type out of the package we should be unit testing this.

Reply:
added unit tests.